src/Case-Channel.f90

```diff
@@ -274,19 +260,111 @@ subroutine channel_cfr (ux, constant)
     enddo
 
   end subroutine channel_cfr
+
   !############################################################################
   !############################################################################
   subroutine postprocess_channel(ux1,uy1,uz1,pp3,phi1,ep1)
 
-    use var, ONLY : nzmsize
+    use var, ONLY : nzmsize,xnu
+    USE variables, only: nx,ny,nz
+
+    use MPI
 
     implicit none
 
     real(mytype), intent(in), dimension(xsize(1),xsize(2),xsize(3)) :: ux1, uy1, uz1, ep1
     real(mytype), intent(in), dimension(xsize(1),xsize(2),xsize(3),numscalar) :: phi1
     real(mytype), intent(in), dimension(ph1%zst(1):ph1%zen(1),ph1%zst(2):ph1%zen(2),nzmsize,npress) :: pp3
 
+    real(mytype) :: ufriction,umax,resdux1,resduy1,resduz1,temp1,temp2
+    integer :: nxc, nyc, nzc, xsize1, xsize2, xsize3
+    integer :: i,j,k
+    integer :: code
+
+    if (nclx1==1 .and. xend(1)==nx) then
+       xsize1=xsize(1)-1
+    else
+       xsize1=xsize(1)
+    endif
+    if (ncly1==1 .and. xend(2)==ny) then
+       xsize2=xsize(2)-1
+    else
+       xsize2=xsize(2)
+    endif
+    if (nclz1==1 .and. xend(3)==nz) then
+       xsize3=xsize(3)-1
+    else
+       xsize3=xsize(3)
+    endif
+    if (nclx1==1) then
+       nxc=nxm
+    else
+       nxc=nx
+    endif
+    if (ncly1==1) then
+       nyc=nym
+    else
+       nyc=ny
+    endif
+    if (nclz1==1) then
+       nzc=nzm
+    else
+       nzc=nz
+    endif
+
+    !we only collect statistics every 10 time steps to save computational time
+    if (mod(itime, 10) == 0) then
+
+       temp1=zero
+
+       if (ncly1==2 .and. xstart(2)==1) then
+
+         ! bottom wall, one node away from wall
+         j=2
+         do k=1,xsize(3)
+             do i=1,xsize(1)
+              temp1=temp1+ux1(i,j,k)*ppy(j-1)
+             enddo
+         enddo
+
+       endif
+       !
+       if (nclyn==2 .and. xend(2)==ny) then
+
+         ! upper wall, one node away from wall
+         j=xsize(2)-1
+         do k=1,xsize(3)
+             do i=1,xsize(1)
+              temp1=temp1-ux1(i,j,k)*ppy(j+1)
```

Comment:
`ppy(ny-1)`

Reply:
to be fixed?

Reply:
This has been removed since was part of the postprocessing